dressForTheWeather/View Models/RecommendationViewModel.swift

```diff
@@ -86,15 +77,56 @@ final class RecommendationViewModel {
         )
     }
     
-    private func generateRecommendation(for lowTemp: Double, highTemp: Double, from items: [ClothingItem]) -> [ClothingItem] {
+    private func selectClothingItems(for lowTemp: Double, highTemp: Double, from items: [ClothingItem]) -> [ClothingItem] {
         return items.filter { $0.tempRange.contains(lowTemp) || $0.tempRange.contains(highTemp) }
     }
     
     private func setRecommendations(for lowTemp: Double, highTemp: Double) {
-        let recommendedItems = generateRecommendation(for: lowTemp, highTemp: highTemp, from: allClothingItems)
-        recommendedClothingItems = recommendedItems
-        let outfit = Outfit(components: recommendedItems)
-        recommendations = outfit.recommendations
+        let recommendedItems = selectClothingItems(for: lowTemp, highTemp: highTemp, from: allClothingItems)
+        
+        recommendedClothingItems = []
+        
+        BodyPlacement.allCases.forEach { currentPlacement in
+            let itemsWithPlacement = recommendedItems.filter { $0.placement.contains(currentPlacement) }
+            
+            guard let randomItem = itemsWithPlacement.randomElement(),
+                recommendedClothingItems.filter({ $0.placement.contains(currentPlacement) }).isEmpty else { return }
+            
+            recommendedClothingItems.append(randomItem)
+        }
+        
+        let outfit = Outfit(components: recommendedClothingItems)
+        recommendations = outfit.recommendationString
+    }
+    
+    public func updateBackgroundColors() -> [CGColor]? {
+        let colorRanges = getColorTemperatureRanges(from: highTemp)
+        
+        guard let color1 = UIColor(named: colorRanges.range1.rawValue),
+              let color2 = UIColor(named: colorRanges.range2.rawValue) else { return nil }
+        
+        let colors = [color1.cgColor, color2.cgColor]
+        return colors
+    }
+    
+    func getColorTemperatureRanges(from temp: Double) -> (range1: TemperatureRanges, range2: TemperatureRanges) {
```

Comment:
I think this could live in the TemperatureRanges enum as a static function (as I mentioned in above comments). Then it could be called like this: TemperatureRange.from(temp: highTemp)

Reply:
yes, this is so much better. Done in 79ce1cfcdeec7ac77adfa4402d2d3b6636544a74